src/PrimeWorker.java

```diff
@@ -1,19 +1,34 @@
+import java.math.BigInteger;
 
 public class PrimeWorker implements Runnable {
 	private long num;
-	
+	private BigInteger bigNum;
+	boolean usingBigNum = false;
+
 	//This constructor is where the PrimeWorker object gets assigned its number
 	public PrimeWorker(long num){
 		this.num = num;
```

Comment:
It would be easier to just call the other constructor instead

That would minimize your added code and simplify.